### PR TITLE
Minor Fine-Tuning Scheduler Tutorial Updates for Lightning 1.8

### PR DESCRIPTION
**Pull request by speediedan**
## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## What does this PR do?

Two minor updates to the Fine-Tuning Scheduler tutorial to better support the release of Lightning 1.8.x (supported by Fine-Tuning Scheduler 0.3.x)

- removed use of a deprecated HF datasets call
- added support for DDP_FORK (and associated alias strategies) though still using a single device by default for colab

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@speediedan seems we have finally fixed the ci/cd and so we are fine to proceed... :rabbit:

File: lightning_examples/finetuning-scheduler/.meta.yml [hunk 1]
```diff
@@ -14,6 +14,6 @@ description: |
   schedule. It uses Hugging Face's ``datasets`` and ``transformers`` libraries to retrieve the relevant benchmark data
   and foundational model weights. The required dependencies are installed via the finetuning-scheduler ``[examples]`` extra.
 requirements:
-  - finetuning-scheduler[examples]>=0.2.0
+  - finetuning-scheduler[examples]>=0.3.0
```
Comment (Borda) [inline lightning_examples/finetuning-scheduler/.meta.yml:17]:
```suggestion
  - finetuning-scheduler[examples]>=0.3.0
  - datasets<2.8.0  # todo: AttributeError: module 'datasets.arrow_dataset' has no attribute 'Batch'
```

Comment (Borda) [reply lightning_examples/finetuning-scheduler/.meta.yml:17]: @speediedan maybe need an update also in `finetuning` :otter:

**Review by Borda — APPROVED** [review]

File: lightning_examples/finetuning-scheduler/.meta.yml (hunk 1 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/.meta.yml:17]: good catch man! Delighted the CI issue was resolved. Looks like 1.9 is on the way so I'll ensure the tutorial is updated appropriately for that version as soon as I have a chance as well.

**Pull request merged**